saskatoon/harvest/api.py

```diff
@@ -197,28 +198,71 @@ def list(self, request, *args, **kwargs):
         return Response({'data': response.data})
 
 
-class BeneficiaryViewset(LoginRequiredMixin, viewsets.ModelViewSet):
-    """Beneficiary viewset"""
+class OrganizationViewset(LoginRequiredMixin, viewsets.ModelViewSet):
+    """
+    Organization viewset
+    - shared detail page for organizations
+    - separate list pages for beneficiaries and equipment points.
+    """
 
     permission_classes = [IsPickLeaderOrCoreOrAdmin]
     queryset = Organization.objects.all().order_by('-actor_id')
-    serializer_class = BeneficiarySerializer
+    serializer_class = OrganizationSerializer
     filter_backends = (filters.DjangoFilterBackend,)
     filterset_class = OrganizationFilter
-    template_name = 'app/list_views/beneficiary/view.html'
 
-    def list(self, request, *args, **kwargs):
-        response = super(BeneficiaryViewset, self).list(request, *args, **kwargs)
-        if request.accepted_renderer.format == 'json':
+    def retrieve(self, request, format='html', pk=None):
+        self.template_name = 'app/detail_views/organization/view.html'
+
+        pk = self.get_object().pk
+        response = super(OrganizationViewset, self).retrieve(request, pk=pk)
+
+        if format == 'json':
             return response
+
         # default request format is html:
-        return Response({'data': response.data,
-                         'filter': get_filter_context(self),
-                         'new': {'url': reverse_lazy('beneficiary-create'),
-                                 'title': _("New Organization")
-                                 }
+        return Response({'organization': response.data,
+                         'data': Equipment.objects.filter(owner_id=pk)
                          })
 
+    @action(detail=False, methods=['get'])
+    def beneficiary_list(self, request, *args, **kwargs):
+        self.template_name = 'app/list_views/beneficiary/view.html'
+        self.filterset_class = BeneficiaryFilter
+
+        response = super(OrganizationViewset, self).list(request, *args, **kwargs)
+
+        if request.accepted_renderer.format == 'json':
+            return response
+
+        return Response({
+            'data': response.data,
+            'filter': get_filter_context(self),
+            'new': {
+                'url': reverse_lazy('organization-create'),
+                'title': _("New Organization")
+            }
+        })
+
+    @action(detail=False, methods=['get'])
+    def equipment_point_list(self, request, *args, **kwargs):
+        self.template_name = 'app/list_views/equipment_point/view.html'
+        self.filterset_class = EquipmentPointFilter
+
+        response = super(OrganizationViewset, self).list(request, *args, **kwargs)
```

Comment:
I think we should keep things more simple and create 2 separate `ListAPIView`s + 1 `DetailAPIView`. This way each list view can clearly define it's filter and it's initial queryset. Example:
https://github.com/LesFruitsDefendus/saskatoon-ng/blob/e8ac1756095791359eacff3e1207cd22002d67b4/saskatoon/harvest/api.py#L251

Reply:
I imagined the queryset to be identical no? that was the main reason i switched gears and chose to keep it all on the same Viewset. if not, it wouldn't make much sense to have a filters like `is_equipment_point` for the Equipment Points list.

these two lists can still override the base classes `self.queryset`. not sure what the three separate views have over this, though it wouldn't be too hard to change